GeneratorInterface/Pythia8Interface/interface/Py8HMC3InterfaceBase.h

```diff
@@ -37,8 +37,8 @@ namespace gen {
     virtual void statistics();
     virtual const char* classname() const = 0;
 
-    void p8SetRandomEngine(CLHEP::HepRandomEngine* v) { p8RndmEngine_.setRandomEngine(v); }
-    P8RndmEngine& randomEngine() { return p8RndmEngine_; }
+    void p8SetRandomEngine(CLHEP::HepRandomEngine* v) { p8RndmEngine_->setRandomEngine(v); }
+    P8RndmEnginePtr randomEngine() { return p8RndmEngine_; }
```

Comment:
By quick look it didn't seem the callers of `randomEngine()` would need the `shared_ptr` access, and this function could stay to return `P8RndmEngine&`
```suggestion
    P8RndmEngine& randomEngine() { return *p8RndmEngine_; }
```
Or did I miss something?